src/main/java/view/ListView.java

```diff
@@ -318,9 +355,68 @@ public void clearTable() {
 
     private void setRecommendationFields(ListState state) {
         recommendBox.setText(state.getGeneratedRecommendations());
+        recommendBox.updateUI();
     }
 
-    public void setListController(ListController listController) {
-        this.listController = listController;
+    /**
+     * A listener for rating column updates in the table.
+     */
+    class UserRatingUpdateListener implements TableModelListener {
```

Comment:
What would you think about moving this class to its own java file? Our list view is pretty cluttered as is, and while this is relevant to its functionality, I think it's distinct enough to deserve it's own file.

Reply:
Yeah, we could do that, this doesn't have to be an inner class.

Reply:
I had to make a few getters and a setter for some private variables in the View Class, I hope this won't come back to haunt us later.